WordPress/Classes/ViewRelated/Notifications/Views/NoteBlockHeaderTableViewCell.swift

```diff
@@ -2,97 +2,37 @@ import Foundation
 import WordPressShared.WPStyleGuide
 import WordPressUI
 import Gravatar
+import SwiftUI
+import DesignSystem
 
 // MARK: - NoteBlockHeaderTableViewCell
 //
 class NoteBlockHeaderTableViewCell: NoteBlockTableViewCell {
-
-    // MARK: - Private
-    private var authorAvatarURL: URL?
-    private typealias Style = WPStyleGuide.Notifications
-
-    // MARK: - IBOutlets
-    @IBOutlet private var authorAvatarImageView: UIImageView!
-    @IBOutlet private var headerTitleLabel: UILabel!
-    @IBOutlet private var headerDetailsLabel: UILabel!
-
-    // MARK: - Public Properties
-    @objc var headerTitle: String? {
-        set {
-            headerTitleLabel.text  = newValue
-        }
-        get {
-            return headerTitleLabel.text
-        }
-    }
-
-    @objc var attributedHeaderTitle: NSAttributedString? {
-        set {
-            headerTitleLabel.attributedText  = newValue
-        }
-        get {
-            return headerTitleLabel.attributedText
-        }
-    }
-
-    @objc var headerDetails: String? {
-        set {
-            headerDetailsLabel.text = newValue
-        }
-        get {
-            return headerDetailsLabel.text
-        }
-    }
-
-    @objc var attributedHeaderDetails: NSAttributedString? {
-        set {
-            headerDetailsLabel.attributedText  = newValue
-        }
-        get {
-            return headerDetailsLabel.attributedText
-        }
+    private var hostViewController: UIHostingController<HeaderView>?
+
+    func configure(text: String, avatarURL: URL?, action: @escaping () -> Void) {
+        let headerView = HeaderView(image: avatarURL, text: text, action: action)
+        hostViewController = UIHostingController(rootView: headerView)
+        guard let hostViewController = hostViewController else { return }
+        hostViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        contentView.addSubview(hostViewController.view)
+        contentView.pinSubviewToAllEdges(hostViewController.view)
     }
+}
```

Comment:
[Suggestion] While this code works well, there’s room to enhance how the view is hosted in UIKit:

1. Ensure the `UIHostingController` is added as a child view controller. This step integrates it properly into the lifecycle of the parent controller.
2. If `hostViewController` is already initialized, consider updating its `rootView` within the `configure` method. Since `configure` could be invoked multiple times due to cell reuse, updating the existing `hostViewController` instead of leaving it unchanged will help keep your UI consistent and responsive to changes.

You can see a similar implementation in `HostingTableViewCell.swift` file.